system/jlib/jfile.cpp

```diff
@@ -2055,11 +2059,19 @@ CFileIO::CFileIO(HANDLE handle, IFOmode _openmode, IFSHmode _sharemode, IFEflags
     sharemode = _sharemode;
     openmode = _openmode;
     extraFlags = _extraFlags;
+
+    // leave for compatibility
     if (extraFlags & IFEnocache)
         if (!isPCFlushAllowed())
             extraFlags = static_cast<IFEflags>(extraFlags & ~IFEnocache);
+
+    if (hasExpertFileFlags == POS_MASK)
+        extraFlags = static_cast<IFEflags>(extraFlags | expertFileFlagsMask);
+    else if (hasExpertFileFlags == NEG_MASK)
+        extraFlags = static_cast<IFEflags>(extraFlags & expertFileFlagsMask);
```

Comment:
see other comments, maybe simpler if have a expertEnableIFileFlagsMask expertDisableIFileFlagsMask, that includes the IFEnocache (if !isPCFlushAllowed).
Then this becomes:
```
    extraFlags = _extraFlags;
    extraFlags = static_cast<IFEflags>(extraFlags | expertEnableIFileFlagsMask);
    extraFlags = static_cast<IFEflags>(extraFlags & ~expertDisableIFileFlagsMask);
```